src/core/Tm.ml

```diff
@@ -10,6 +10,37 @@ type bface = (tm, btm) Desc.Boundary.face
 type bsys = (tm, btm) Desc.Boundary.sys
 type data_desc = (tm, btm) Desc.desc
 
+type ('a, 'k) telescope =
+  | TNil of 'k
+  | TCons of 'a * ('a, 'k) telescope bnd
+
+
+module NewDesc =
+struct
+  type btm =
+    | Var of int
+    | Intro of
+        {clbl : string;
+         const_args : tm list;
+         rec_args : btm list;
+         rs : tm list}
+
+  type bface = tm * tm * btm
```

Comment:
Warning: bface will be "some flags" (like cleanness) * tm * tm * btm in the future.

Reply:
I am deciding now, I think, that it is way too much work to maintain these separate formal boundary terms (because of elaboration), and that it will be way easier to just use ordinary terms and make sure they are of the right shape.

Please stay tuned.

Reply:
That is good because I will request wild things later, for example, the boundaries with recursors on previously defined types!

Reply:
I'm glad you agree.... Otherwise we'd have to just replicate more and more of the language except in a more complicated way. what a horrible prospect!